google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -253,6 +254,40 @@ public static BucketSourceOption userProject(String userProject) {
     }
   }
 
+  /** Class for specifying listHmacKeys options */
+  class ListHmacKeysOption extends Option {
+    private ListHmacKeysOption(StorageRpc.Option rpcOption, Object value) {
+      super(rpcOption, value);
+    }
+
+    /**
+     * Returns an option for the Service Account whose keys to list. If this option is not used,
+     * keys for all accounts will be listed.
+     */
+    public static ListHmacKeysOption serviceAccount(ServiceAccount serviceAccount) {
+      return new ListHmacKeysOption(
+          StorageRpc.Option.SERVICE_ACCOUNT_EMAIL, serviceAccount.getEmail());
+    }
+
+    /** Returns an option for the maximum amount of HMAC keys returned per page. */
+    public static ListHmacKeysOption maxResults(long pageSize) {
+      return new ListHmacKeysOption(StorageRpc.Option.MAX_RESULTS, pageSize);
+    }
+
+    /** Returns an option to specify the page token from which to start listing HMAC keys. */
+    public static ListHmacKeysOption pageToken(String pageToken) {
+      return new ListHmacKeysOption(StorageRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to specify whether to show deleted keys in the result. This option is false
+     * by default.
+     */
+    public static ListHmacKeysOption showDeletedKeys(boolean showDeletedKeys) {
```

Comment:
`userProject` should be available based on the discovery document. https://www.googleapis.com/discovery/v1/apis/storage/v1/rest